### ISSUE TITLE
UPDATE MISSOURI SUBSCRIPTION -reopened on May 28, 2025

### ISSUE THREAD

**Issue by mnichols13** [MyCaert/Issues-Enhancements#306]
Missouri has extended their subscription for another year. Please make the following changes for all Missouri ag teachers ASAP

For any teachers with a master de-activate date of 7/1/20, we need you to change it to 7/1/21.

For any teachers who have de-activate dates of 7/1/20 to the six Missouri libraries (Agriscience, Horticulture, Animals, Plants & Soils, Ag Mechanics, Agribusiness, and Environmental Resources) we need you to change them to 7/1/2021. This includes lesson plan, Eunit, and CSAT access for the six libraries.

(refer to 245 closed issue)

**Comment by bguerrettaz** [comment]
Hi - this is complete, please confirm.

Thanks,
Blake

**Comment by mnichols13** [comment]
looks good!  thank you

**Comment by mnichols13** [comment]
There appears to be a very large number of Missouri teachers that see no lesson libraries when they login.   They can access the lesson libraries if they have created a course outline and saved it, however they cannot see anything on the homepage.    We need this to be corrected tonight if possible please. 
![image](https://user-images.githubusercontent.com/5790342/88973632-fec01400-d27c-11ea-8769-692742f0ee9a.png)

**Comment by dpentony** [comment]
I just noticed that if you go into Edit the teacher account and simply click save changes (without changing anything else), then the libraries will show up when the teachers login. I experimented on another teacher (Caleb Stewart - Neelyville) and it worked.

Not sure if that helps in troubleshooting the issues.

**Comment by bguerrettaz** [comment]
That does help, thank you.  What I found was the first teacher had a blank
LessonPlanArray in the database and Caleb had the appropriate CD's listed.
Here are the distinct values in LessonPlanArray for all MO teachers:
[image: image.png]

Row 10 (blank) is what's causing the problem.  A fix may be to put in the
most common values, which is [19, 20, 21, 22, 23, 24]

Please note, when I made the changes for this state a few weeks ago, this
is a column I don't update or change.  I can't speak to why this happened;
as we've mentioned in the past this design and all of the code around it is
very messy.

Please let me know if you'd like me to update all MO teachers in this
column with [19, 20, 21, 22, 23, 24], which should solve the problem for
the MO teachers.  I will backup this table and it can be restored if you
encounter issues in testing.

Thanks,
Blake

**Comment by mnichols13** [comment]
whatever it takes to fix it.. is that what you advice?

**Comment by bguerrettaz** [comment]
Yes, as long as you can test a handful of teachers quickly after I make the
change so I can restore the backup if you find an issue.

**Comment by mnichols13** [comment]
yes just let me know when

**Comment by bguerrettaz** [comment]
Ok, I had the queries written and ready to go.  I've backed up the table
and executed the query.  It seemed to resolve the teachers issue, but
please confirm the rest.

Thanks,
Blake

**Comment by mnichols13** [comment]
looking now

**Comment by mnichols13** [comment]
still not right with the teachers set up under the school:  College of the Ozarks

continuing to check - all others so far look good.

**Comment by mnichols13** [comment]
also problems with Northwest Missouri State
Haley Wessling
Brice Ball
Ashley Layman
Jessica Noker
Courtney Schreier

could be others - but definitely not all of them

**Comment by bguerrettaz** [comment]
Hi - were these all resolved?  I'm seeing the values in that column...or is
this still an issue?  If it still an issue, I might have another idea.

**Comment by mnichols13** [comment]
the teachers that I mentioned in the two previous comments are still in need of a fix

**Comment by bguerrettaz** [comment]
Hi - I believe that I have all of the Northwest Missouri State teachers
corrected.  Can you please test and confirm those?  If you validate, I can
move to College of the Ozarks.

Apologies this takes so long, but as we've discussed many times in the
past, there are many issues with between all of the date/license tables and
I'm fixing as we encounter them.

Blake

**Comment by mnichols13** [comment]
Yes this fixed the NW MO students

**Comment by bguerrettaz** [comment]
College of the Ozarks should be fixed now as well.

Thanks,
Blake

**Comment by mnichols13** [comment]
looks good. thanks.

**Comment by mnichols13** [comment]
Missouri has extended their subscription for another year. Please make the following changes for all Missouri AG teachers ASAP

For any teachers with a master de-activate date of 7/1/21, we need you to change it to 7/1/22.

For any teachers who have de-activate dates of 7/1/21 to the six Missouri libraries (Agriscience, Horticulture, Animals, Plants & Soils, Ag Mechanics, Agribusiness, and Environmental Resources) we need you to change them to 7/1/2022. This includes lesson plan, Eunit, and CSAT access for the six libraries.

**Please complete by end of the day June 29th.**

**Comment by bguerrettaz** [comment]
Hi - MO is complete, please test and confirm.

Thanks,
Blake

**Comment by mnichols13** [comment]
looks good

**Comment by mnichols13** [comment]
Missouri has extended their subscription for another year. Please make the following changes for all Missouri AG teachers

For any teachers with a master de-activate date of 7/1/22, we need you to change it to 7/1/23.

For any teachers who have de-activate dates of 7/1/22 to the six Missouri libraries (Agriscience, Horticulture, Animals, Plants & Soils, Ag Mechanics, Agribusiness, and Environmental Resources) we need you to change them to 7/1/2023. This includes lesson plan, Eunit, and CSAT access for the six libraries.

Desired completion date:  June 28 or before

**Comment by mnichols13** [comment]
Not sure what happened on this one - but it does not look like the teachers updated - still showing they expire 7-1-2022

**Comment by bguerrettaz** [comment]
This is complete.

Thanks,
Blake

**Comment by mnichols13** [comment]
looks good

**Comment by mnichols13** [comment]
@bguerrettaz  this requested completion date was Friday the 23rd of June.  How quickly can you get to this?  

Thanks.

**Comment by bguerrettaz** [comment]
Hi - my bad, I thought this had a 6.28 date on it.  I should have it
completed by tomorrow.

**Comment by mnichols13** [comment]
Thank you

**Comment by bguerrettaz** [comment]
Hi - I will try to get this done by EOD Wednesday.

Blake

**Comment by bguerrettaz** [comment]
Hi - this work is complete.

Thanks,
Blake

**Comment by mnichols13** [comment]
looks good, thanks for your quick response.  I appreciate it!

**Comment by mnichols13** [comment]
@bguerrettaz When could you do this?  Missouri has extended their subscription for another year. Please make the following changes for all Missouri ag teachers .

For any teachers with a master de-activate date of 7/1/25, we need you to change it to 7/1/26.

For any teachers who have de-activate dates of 7/1/25 to the six Missouri libraries (Agriscience, Horticulture, Animals, Plants & Soils, Ag Mechanics, Agribusiness, and Environmental Resources) we need you to change them to 7/1/2026. This includes lesson plan, Eunit, and CSAT access for the six libraries.

**Comment by bguerrettaz** [comment]
I will be able to work on it this week.

Thanks,
Blake

**Comment by mnichols13** [comment]
tested and confirmed that this is completed
